<launch>
  <!-- Serial port for the optoforce sensor -->
  <arg name="port" default="/dev/ttyACM0"/>
  <!-- Update speed. Available values:
        "Stop"
        "1000Hz"
        "333Hz"
        "100Hz"
        "30Hz"
        "10Hz"
        -->
  <arg name="speed" default="100Hz"/>
  <!-- Internal filtering. Available values:
        "No"
        "500Hz"
        "150Hz"
        "50Hz"
        "15Hz"
        "5Hz"
        "1.5Hz"
        -->
  <arg name="filter" default="15Hz"/>
  <!-- Clear the sensors offset -->
  <arg name="zero" default="true"/>
  <!-- DAQ (Data Acquisition) Type. It can be single or multiple channel, and have 3 or 6 axis. Available values:
        "s-ch/3-axis"
        "m-ch/3-axis"
        "s-ch/6-axis"
        -->
  <arg name="type" default="m-ch/3-axis"/>
  <!-- Numerical index of the first sensor. The rest (if this is a multi-channel DAC)
  will get the consecutive numbers -->
  <arg name="starting_index" default="0"/>
  <!-- When append_serial_number is set to true, the topic on which wrenches are
  published will be named after the sensor's serial number. -->
  <arg name="append_serial_number" default="true"/>
  <!-- Direction of scaling file -->
  <arg name="scaling_file" default="$(find optoforce)/config/multi_channel_3_axis_generic_scale.yaml"/>

  <node name="$(anon optoforce)" pkg="optoforce" type="optoforce.py" output="screen">
    <param name="port" value="$(arg port)"/>
    <param name="speed" value="$(arg speed)"/>
    <param name="filter" value="$(arg filter)"/>
    <param name="zero" value="$(arg zero)"/>
    <param name="type" value="$(arg type)"/>
    <param name="starting_index" value="$(arg starting_index)"/>
    <param name="append_serial_number" value="$(arg append_serial_number)"/>
    <rosparam command="load" file="$(arg scaling_file)"/>
  </node>
</launch>
